test/sslapitest.c

```diff
@@ -3735,6 +3735,114 @@ static int test_ciphersuite_change(void)
     return testresult;
 }
 
+/*
+ * Test TLSv1.3 Cipher Suite
+ * Test 0 = Set TLS1.3 cipher on context
+ * Test 1 = Set TLS1.3 cipher on SSL
+ * Test 2 = Set TLS1.3 and TLS1.2 cipher on context
+ * Test 3 = Set TLS1.3 and TLS1.2 cipher on SSL
+ */
+static int test_tls13_ciphersuite(int idx)
+{
+    SSL_CTX *sctx = NULL, *cctx = NULL;
+    SSL *serverssl = NULL, *clientssl = NULL;
+    static const char *t13_ciphers[] = {
+        TLS1_3_RFC_AES_128_GCM_SHA256,
+        TLS1_3_RFC_CHACHA20_POLY1305_SHA256,
+        TLS1_3_RFC_AES_128_CCM_SHA256,
+        TLS1_3_RFC_AES_256_GCM_SHA384":"TLS1_3_RFC_CHACHA20_POLY1305_SHA256,
+        TLS1_3_RFC_AES_128_CCM_8_SHA256":"TLS1_3_RFC_AES_128_CCM_SHA256
+    };
+    const char *t13_cipher = NULL;
+    const char *t12_cipher = NULL;
+    int set_at_ctx = 0;
+    int set_at_ssl = 0;
+    int testresult = 0;
+    int i;
+
+    switch (idx) {
+        case 0:
+            set_at_ctx = 1;
+            break;
+        case 1:
+            set_at_ssl = 1;
+            break;
+        case 2:
+            set_at_ctx = 1;
+            t12_cipher = TLS1_TXT_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256;
+            break;
+        case 3:
+            set_at_ssl = 1;
+            t12_cipher = TLS1_TXT_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256;
+            break;
+    }
+
+    for (i = 0; i < (int)(sizeof(t13_ciphers)/sizeof(t13_ciphers[0])); i++) {
+        t13_cipher = t13_ciphers[i];
+        if (!TEST_true(create_ssl_ctx_pair(TLS_server_method(), TLS_client_method(),
+                                           TLS1_VERSION, TLS1_3_VERSION,
```

Comment:
This line appears different in this port. Why is this not TLS_MAX_VERSION here instead of TLS1_3_VERSION?

Reply:
This is the only change which I have done when I port. In master branch `TLS_MAX_VERSION` macro is under `!OPENSSL_API_3` macro, so I thought this macro will be removed.

Reply:
Ah. Right. In fact the correct answer is to use the value 0 here (which means no maximum). In fact that should also work in 1.1.1, which means you can use the same PR to target both branches.

Reply:
Changed it to `0` on both PRs. I am not able to find option in github PR to make multiple branches as target branch.

Reply:
@mattcaswell Please suggest, can I close the other PR #9064 and keep this one for merging on both 1.1.1 and master or vice versa.

Reply:
I closed the other PR, and added labels to this PR indicating that we are targeting both master and 1.1.1.